plugins/ci-github-actions/src/index.ts

```diff
@@ -0,0 +1,147 @@
+import type {
+  AmplicationPlugin,
+  CreateServerParams,
+  DsgContext,
+  Events,
+  Module,
+} from "@amplication/code-gen-types";
+import {
+  authenticationPasswordKey,
+  authenticationUsernameKey,
+  imageKey,
+  serviceWorkingDirectoryKey,
+  serviceNameKey,
+} from "./constants";
+import { getPluginSettings } from "./utils";
+import { EventNames } from "@amplication/code-gen-types";
+import { resolve } from "path";
+
+class GithubActionsPlugin implements AmplicationPlugin {
+  register(): Events {
+    return {
+      [EventNames.CreateServer]: {
+        after: this.afterCreateServer,
+      },
+    };
+  }
+
+  async afterCreateServer(
+    context: DsgContext,
+    eventParams: CreateServerParams,
+    modules: Module[]
+  ) {
+    context.logger.info(`Generating Github Actions workflow...`);
+
+    // determine the name of the service which will be used as the name for the workflow
+    // workflow names must be lower case letters and numbers. words may be separated with dashes (-):
+    const serviceName = context.resourceInfo?.name
+      .toLocaleLowerCase()
+      .replaceAll(/[^a-zA-Z0-9-]/g, "-");
```

Comment:
Isn't it better to use the kebabCase() method from lodash as we do in other places in the code?

Reply:
moved the setup of the service name over to kebabCase for consistency